app/controllers/application_controller.rb

```diff
@@ -1,2 +1,18 @@
 class ApplicationController < ActionController::API
+  before_action :authenticate
+
+  private
+
+  def login
+    if (session[:token])
+      @current_user = User.find_by(token: session[:token])
+    else
+      @current_user ||= User.find_by(token: params[:token])
+      session[:token] = params[:token]
```

Comment:
Most api users expect to pass auth tokens with each request, either in params or headers so you probably don't need session storage - its also possible that whatever client they use to make the requests (usually not a web browser) would not have cookie/session storage anyways